src/components/matrix/virtualized-grid.tsx

```diff
@@ -0,0 +1,408 @@
+
+  return (
+    <div className='grid-container' style={gridContainerStyles}>
+      <RowHeaders
+        ref={rowLabelRef}
+        cellHeight={cellHeight}
+        cellWidth={cellWidth}
+        width={rowHeaderWidth}
+        top={columnHeaderHeight}
+        height={gridHeight}
+        itemCount={numRows}
+        itemData={rowHeaderData}
+        onScroll={handleRowLabelScroll}
+      />
+      <ColumnHeaders
+        ref={columnLabelRef}
+        height={columnHeaderHeight}
+        cellWidth={cellWidth}
+        left={rowHeaderWidth}
+        width={gridWidth}
+        itemCount={numColumns}
+        itemData={columnHeaderData}
+        onScroll={handleColumnLabelScroll}
+      />
+      <div className='grid-test'>
+        <FixedSizeGrid
+          className='grid'
+          style={gridStyles}
+          width={gridWidth - 15}
+          height={gridHeight - 15}
+          columnWidth={cellWidth}
+          rowHeight={cellHeight}
+          rowCount={numRows}
+          columnCount={numColumns}
+          itemData={gridData}
+          onScroll={handleGridScroll}
+          ref={gridRef}
+          overscanColumnCount={10}
+          overscanRowCount={10}
+          itemKey={gridItemKey}
+        >
+          {GridCell}
+        </FixedSizeGrid>
+      </div>
+      {showRight && <GridRightButton onClick={pressScrollRight} rowHeaderWidth={rowHeaderWidth} />}
+      {showUp && <GridUpButton onClick={pressScrollUp} rowHeaderWidth={rowHeaderWidth} />}
+      {showLeft && <GridLeftButton onClick={pressScrollLeft} rowHeaderWidth={rowHeaderWidth} />}
+      {showDown && <GridDownButton onClick={pressScrollDown} rowHeaderWidth={rowHeaderWidth} />}
+    </div>
+  );
+};
+
+type GridMoveButton = {
+  onClick: MouseEventHandler;
+  rowHeaderWidth: number;
+};
+
+const GridLeftButton = ({ onClick, rowHeaderWidth }: GridMoveButton): JSX.Element => {
+  return (
+    <button
+      title='Scroll Left Button'
+      className='grid-left-btn'
+      onClick={onClick}
+      style={{
+        display: 'flex',
+        alignItems: 'center',
+        backgroundColor: 'white',
+        height: 20,
+        position: 'absolute',
+        top: 12,
+        left: rowHeaderWidth - 30,
+      }}
+    >
+      <svg
+        xmlns='http://www.w3.org/2000/svg'
+        width='20'
+        height='20'
+        fill='#4674a7'
+        className='bi bi-chevron-double-left'
+        viewBox='0 0 16 16'
+      >
+        <path
+          fillRule='evenodd'
+          d='M8.354 1.646a.5.5 0 0 1 0 .708L2.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z'
+        />
+        <path
+          fillRule='evenodd'
+          d='M12.354 1.646a.5.5 0 0 1 0 .708L6.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z'
+        />
+      </svg>
+    </button>
+  );
+};
+
+const GridUpButton = ({ onClick, rowHeaderWidth }: GridMoveButton): JSX.Element => {
+  return (
+    <div
```

Comment:
I think you mentioned that you've borrowed these from bootstrap icons. So please add a link to the source